### PR TITLE
Lps 196762

### PR DESCRIPTION
**Pull request by SamZiemer**
https://liferay.atlassian.net/browse/LPS-196762

Image magick on windows does not work with the convert command. Instead we must use magick instead.

/cc @manuelRives, @robertoDiaz

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

**Comment by SamZiemer** [comment]
ci:test

**Comment by SamZiemer** [comment]
ci:test:relevant

File: modules/apps/image/image-impl/src/main/java/com/liferay/image/internal/ImageMagickImpl.java [hunk 1]
```diff
@@ -226,7 +227,16 @@ public byte[] scale(byte[] bytes, String mimeType, int width, int height)
...
+			if (_log.isDebugEnabled()) {
+				ProcessEvent processEvent = (ProcessEvent)future.get();
+
+				if (processEvent.getException() != null) {
+					_log.debug(processEvent.getException());
+				}
+			}
+			else {
+				future.get();
+			}
```
Comment (adolfopa) [inline modules/apps/image/image-impl/src/main/java/com/liferay/image/internal/ImageMagickImpl.java:239]: Log blocks should not contain logic not related to logging. Can you refactor to something like this?
```
ProcessEvent processEvent = (ProcessEvent)future.get();

if (_log.isDebugEnabled() && (processEvent.getException() != ...)) {
     _log.debug(...);
}
```
This way, it is more clear that we can remove the debug statements without affecting the behavior.

**Review by adolfopa — CHANGES_REQUESTED** [review]

File: modules/apps/image/image-impl/src/main/java/com/liferay/image/internal/ImageMagickImpl.java [hunk 2]
```diff
@@ -226,7 +227,11 @@ public byte[] scale(byte[] bytes, String mimeType, int width, int height)
 
 			Future<?> future = convert(arguments);
 
-			future.get();
+			ProcessEvent processEvent = (ProcessEvent)future.get();
+
+			if (_log.isDebugEnabled() && processEvent.getException() != null) {
```
Comment (adolfopa) [inline modules/apps/image/image-impl/src/main/java/com/liferay/image/internal/ImageMagickImpl.java:232]: There's an SF issue here; the second subexpression should be bracketed `... && (processEvent.getException() ...))`

**Review by adolfopa — CHANGES_REQUESTED** [review]

File: modules/apps/image/image-impl/src/main/java/com/liferay/image/internal/ImageMagickImpl.java (hunk 2 above)
Comment (SamZiemer) [reply modules/apps/image/image-impl/src/main/java/com/liferay/image/internal/ImageMagickImpl.java]: Thanks @adolfopa, sorry about that. I've updated it now.

**Review by adolfopa — APPROVED** [review]

**Comment by adolfopa** [comment]
ci:forward

**Comment by SamZiemer** [comment]
ci:test:relevant

**Comment by Jeremy-Ch** [comment]
The failures are unrelated.

**Comment by Jeremy-Ch** [comment]
ci:forward

**Comment by Jeremy-Ch** [comment]
ci:forward:force

**Comment by Jeremy-Ch** [comment]
ci:test:stable

**Comment by Jeremy-Ch** [comment]
ci:forward:force

**Pull request closed without merging**